File: solutions/ruby/high-scores/1/high_scores.rb
class HighScores
  attr_reader :scores

  def initialize(scores)
      @scores = scores
  end

  def latest
    @scores.at(-1)
  end

  def personal_best
    @scores.max
  end

  def personal_top_three
    @scores.sort.reverse[0..2]
  end

  def latest_is_personal_best?
    latest == personal_best
  end
end

File: solutions/python/high-scores/1/high_scores.py
class HighScores:
    def __init__(self, scores: list[int]):
        self._scores = scores or []
        pass

    def latest(self) -> int:
        return self._scores[-1]

    def personal_best(self) -> int:
        return max(self._scores)

    def personal_top_three(self) -> list[int]:
        return sorted(self._scores)[-3:][::-1]

    @property
    def scores(self) -> list[int]:
        return self._scores
